LayerClient/LayerClient.py

```diff
@@ -0,0 +1,519 @@
+            [
+                MessagePart.from_dict(part) for part in dict_data.get('parts')
+            ],
+        )
+
+    def __repr__(self):
+        return '<LayerClient.Announcement "{text}">'.format(
+            text=self.uuid()
+        )
+
+
+class Message(BaseLayerResponse):
+    """
+    The response returned by the API when a message is sent.
+    """
+
+    def __init__(self, id, url, sent_at=None, sender=None,
+                 conversation=None, parts=None, recipient_status=None):
+        self.id = id
+        self.url = url
+        self.sent_at = sent_at
+        self.sender = sender
+        self.conversation = conversation
+        self.parts = parts
+        self.recipient_status = recipient_status
+
+    @staticmethod
+    def from_dict(dict_data):
+        return Message(
+            dict_data.get('id'),
+            dict_data.get('url'),
+            Message.parse_date(dict_data.get('sent_at')),
+            Sender.from_dict(dict_data.get('sender')),
+            Conversation.from_dict(dict_data.get('conversation')),
+            [
+                MessagePart.from_dict(part) for part in dict_data.get('parts')
+            ],
+            dict_data.get('recipient_status'),
+        )
+
+    def __repr__(self):
+        return '<LayerClient.Message "{uuid}">'.format(
+            uuid=self.uuid()
+        )
+
+
+class Sender:
+    """
+    Used for sending messages.
+    Id and Name may both be set, but the send_message API will prefer
+    one over the other.
+    """
+
+    def __init__(self, id=None, name=None):
+        self.id = id
+        self.name = name
+
+    @staticmethod
+    def from_dict(json):
+        if not json:
+            return None
+
+        return Sender(
+            json.get('id'),
+            json.get('name'),
+        )
+
+    def __repr__(self):
+        return '<LayerClient.Sender "{name}" (id: {id})>'.format(
+            id=self.id,
+            name=self.name,
+        )
+
+    def as_dict(self):
+        # If both ID and name are set, we will default to only the ID.
+        # The layer platform explicitly prohibits sending both.
+        if self.id:
+            return {
+                'user_id': self.id,
+            }
+        else:
+            return {
+                'name': self.name,
+            }
+
+
+class MessagePart:
+    """
+    A message chunk, as used for sending messages.
+
+    Message chunks are currently limited to 2KiB by Layer. If a message is
+    larger, it must be broken into several chunks.
+    By default, chunks are text/plain but can be any format.
+    Messages that are non-text (e.g. images) can be sent as base64. In this
+    case, the encoding field must be set.
+    """
+
+    def __init__(self, body, mime=MIME_TEXT_PLAIN, encoding=None):
+        self.body = body
+        self.mime_type = mime
+        self.encoding = encoding
+
+    @staticmethod
+    def from_dict(dict_data):
+        return MessagePart(
+            dict_data.get('body'),
```

Comment:
Same question as above.  To me the question just comes down to, do you expect to see all of these properties in the response, and if not, where should we handle these errors? In my opinion, I find it easier to catch it as early as possible, but not sure what we expect here